pytensor/tensor/basic.py

```diff
@@ -255,7 +255,26 @@ def _obj_is_wrappable_as_tensor(x):
 )
 
 
-def get_scalar_constant_value(
+def get_scalar_constant(v, elemwise=True, only_process_constants=False, max_recur=10):
+    """
+    Checks whether 'v' is a scalar (ndim = 0).
+
+    If 'v' is a scalar then this function fetches the underlying constant by calling
+    'get_underlying_scalar_constant()'.
+
+    If 'v' is not a scalar, it raises a NotScalarConstantError.
+
+    """
+    if isinstance(v, np.ndarray):
+        data = v.data
+        if data.ndim != 0:
+            raise NotScalarConstantError()
+    return get_underlying_scalar_constant(
+        v, elemwise, only_process_constants, max_recur
+    )
+
+
+def get_underlying_scalar_constant(
```

Comment:
To match the non-underlying 

```suggestion
def get_underlying_scalar_constant_value(
```